File: node/test/test_station.js
var assert = require('assert');
const Station = require('../src/station.js');
const Point = require('../src/point.js');

describe('Station', function() {
  describe('getDistance()', function() {
    it('should return 0 when the point is the same as the location', () => {
      const station = new Station(new Point(0, 0), 10);
      const point = new Point(0, 0);
      assert.equal(station.getDistance(point), 0);
    });
    it('should return the correct distance when the point is not the same as the location', () => {
      const station = new Station(new Point(0, 0), 10);
      const point = new Point(10, 10);
      assert.equal(station.getDistance(point).toFixed(4), 14.1421);
    });
    it('should return the correct distance when the point is not the same as the location and is not zeroed', () => {
        const station = new Station(new Point(5, 5), 10);
        const point = new Point(10, 10);
        assert.equal(station.getDistance(point).toFixed(4), 7.0711);
      });
  });
  describe('getPower()', function() {
    it('should return the full power when device location is the same as the station location', () => {
      const station = new Station(new Point(0, 0), 10);
      const deviceDistance = 0;
      assert.equal(station.getPower(deviceDistance), 100);
    });
    it('should return the correct power when the device location is nearby the station location', () => {
      const station = new Station(new Point(0, 0), 10);
      const deviceDistance = 3;
      assert.equal(station.getPower(deviceDistance).toFixed(4), 49.0000);
    });
    it('should return 0 when the device location is out of reach', () => {
        const station = new Station(new Point(0, 0), 10);
        const deviceDistance = 20;
        assert.equal(station.getPower(deviceDistance), 0);
      });
  });
});

File: python/tests/test_station.py
from src.point import Point
from src.station import Station

def test_same_point_zero_distance():
    station = Station(Point(0, 0), 10)
    p2 = Point(0, 0)
    assert station.distance(p2) == 0

def test_different_points_correct_distance():
    station = Station(Point(0, 0), 10)
    p2 = Point(10, 10)
    assert round(station.distance(p2), 4) == 14.1421

def test_not_zeroed_points_correct_distance():
    station = Station(Point(5, 5), 10)
    p2 = Point(10, 10)
    assert round(station.distance(p2), 4) == 7.0711

def test_location_full_power():
    station = Station(Point(0, 0), 10)
    device_location = Point(0, 0)
    assert station.power(device_location) == 100

def test_nearby_location_correct_power():
    station = Station(Point(0, 0), 10)
    device_location = Point(5, 5)
    assert round(station.power(device_location), 4) == 8.5786

def test_out_of_distance_location_no_power():
    station = Station(Point(0, 0), 10)
    device_location = Point(11, 11)
    assert station.power(device_location) == 0